src/orion/executor/dask_backend.py

```diff
@@ -42,10 +81,31 @@ def wait(self, futures):
         results = self.client.gather(list(futures))
         if self.in_worker:
             rejoin()
+        return [r.get() for r in results]
+
+    def async_get(self, futures, timeout=0.01):
+        results = []
+        tobe_deleted = []
+
+        for future in futures:
+            if timeout:
+                future.wait(timeout)
+
+            if future.ready():
+                try:
+                    results.append(AsyncResult(future, future.get()))
```

Comment:
if `future.future.result` raises an error, would that mean that `future.wait()` above would raise the error before it can be caught here?

Reply:
yeah, good catch, I updated to `wait` captures the exception and it is only thrown when `.get` is called

Reply:
There should be a test for this behavior.

Reply:
added the test `test_execute_async_exception` that makes sure async_get cannot raise 
async_get is the one that calls Future.wait()